### DISCUSSION TITLE
Is there an OpenAPI document for the Keygen API?

### DISCUSSION THREAD

**Discussion by BrettJaner** [keygen-sh/community#16] [Q&A] [answered]
Is there an OpenAPI or Swagger document for the Keygen API?  I was hoping to use it to auto-generate some client code and models.

**Comment by ezekg** [comment, marked as answer]
Not at this time. But this is on my roadmap to explore and possibly get figured out later next year.

Edit: if anybody is reading that wants to help with this, let's chat.

**Reply by MaxWaldorf** [reply]
I have done a baseline for openAPI support on a YAML file...
It is a start but how could I contribute so others can take it and improveit?

Is there a way for it to be postedon a repo?

**Reply by ezekg** [reply]
Please share the repo so that others can use it. If it gains usage and is maintained, we can link to it from the README and the website.

Here's an example of Stripe's OpenAPI repo: https://github.com/stripe/openapi

**Reply by MaxWaldorf** [reply]
Here is the repo:
https://github.com/MaxWaldorf/keygen-sh-openAPI
